gemhardware/include/gem/hw/GEMHwDevice.h

```diff
@@ -123,237 +130,378 @@ namespace gem {
-       * readBlock(std::string const& regName)
-       * read from a memory block
+       * @ingroup uhalwrappers
+       * @brief read from a memory block
+       *
+       * @usage
+       *
        * @param regName fixed size memory block to read from
        */
       std::vector<uint32_t> readBlock(std::string const& regName);
 
       /**
-       * readBlock(std::string const& regName, size_t const nWords)
-       * read from a memory block
-       * @param regName memory block to read from
+       * @ingroup uhalwrappers
+       * @brief read from a memory block
+       *
+       * @usage
+       *
+       * @param regName name of memory block to read from
        * @param nWords size of the memory block to read
-       * @retval returns a vector of 32 bit unsigned value
+       *
+       * @retval a vector of 32 bit unsigned values
        */
-      std::vector<uint32_t> readBlock(std::string const& regName,
-                                      size_t      const& nWords);
+      std::vector<uint32_t> readBlock(std::string const& regName, size_t const& nWords);
 
+      /**
+       * @ingroup uhalwrappers
+       * @brief read from a memory block
+       *
+       * @usage
+       *
+       * @param regName name of memory block to read from
+       * @param buffer a pointer to an array containing the read values
+       * @param nWords size of the memory block to read
+       *
+       * @retval the size of the returned array
+       */
       uint32_t readBlock(std::string const& regName, uint32_t* buffer, size_t const& nWords);
-      uint32_t readBlock(std::string const& regName, std::vector<toolbox::mem::Reference*>& buffer,
-                         size_t const& nWords);
+
+      /* / ** */
+      /*  * @ingroup uhalwrappers */
+      /*  * @brief read from a memory block */
+      /*  * */
+      /*  * @usage */
+      /*  * */
+      /*  * @param regName name of memory block to read from */
+      /*  * @param buffer a vector of pointers to memory locations containing the read values */
+      /*  * @param nWords size of the memory block to read */
+      /*  * */
+      /*  * @retval the size of the returned array */
+      /*  * / */
+      /* uint32_t readBlock(std::string const& regName, std::vector<toolbox::mem::Reference*>& buffer, */
+      /*                    size_t const& nWords); */
 
       /**
-       * writeBlock(std::string const& regName, std::vector<uint32_t> const values)
-       * write to a memory block
-       * @param regName memory block to write to
+       * @ingroup uhalwrappers
+       * @brief write to a memory block
+       *
+       * @usage
+       *
+       * @param regName name of memory block to write to
        * @param values list of 32-bit words to write into the memory block
        */
-      void writeBlock(std::string           const& regName,
-                      std::vector<uint32_t> const values);
+      void writeBlock(std::string const& regName, std::vector<uint32_t> const values);
```

Comment:
Why pass by value?

Reply:
Don't remember... probably copied the way it was done where I was taking from had it this way